firmware/quadruna/VC/src/app/states/app_inverterOnState.c

```diff
@@ -4,33 +4,47 @@
 #include "app_canTx.h"
 #include "app_canRx.h"
 #include "app_utils.h"
+#include <stddef.h>
 
 static void inverterOnStateRunOnEntry(void)
 {
     app_canTx_VC_State_set(VC_INVERTER_ON_STATE);
     app_powerManager_setState(POWER_MANAGER_DRIVE);
 }
 
-static void inverterOnStateRunOnTick1Hz(void)
-{
-    app_allStates_runOnTick1Hz();
-}
-
 static void inverterOnStateRunOnTick100Hz(void)
 {
-    app_allStates_runOnTick100Hz();
-}
+    const bool all_states_ok = app_allStates_runOnTick100Hz();
+    // Holds previous start switch position (true = UP/ON, false = DOWN/OFF)
+    // Initialize to true to prevent a false start
+    static bool prev_start_switch_pos = true;
 
-static void inverterOnStateRunOnExit(void) {}
+    const bool curr_start_switch_pos      = app_canRx_CRIT_StartSwitch_get();
+    const bool was_start_switch_pulled_up = !prev_start_switch_pos && curr_start_switch_pos;
+    prev_start_switch_pos                 = curr_start_switch_pos;
+
+    const bool bms_in_drive_state = app_canRx_BMS_State_get() == BMS_DRIVE_STATE;
+    const bool is_brake_actuated  = app_canRx_FSM_BrakeActuated_get();
+
+    if (bms_in_drive_state && is_brake_actuated && was_start_switch_pulled_up && all_states_ok)
+    {
+        // Transition to drive state when start-up conditions are passed (see
+        // EV.10.4.3):
+
+        // TODO: Could not thoroughly validate VC refactor without a working BMS.
+        // Thus, re-test IO, app, and vehicle dynamics before going HV up or driving again.
+        app_stateMachine_setNextState(app_driveState_get());
+    }
```

Comment:
Need to check if BMS is **not** in drive or inverter on or precharge state to then go back to init state

Reply:
More specifically: Only turn off the inverters if BMS is in init state